### PR TITLE
Add cluster-federation CSV

### PR DESCRIPTION
**Pull request by pmorie**
Adds `cluster-federation` as a distinct CSV that supports a cluster-scoped deployment of federation in support of #29.

cc @csrwng

### CODE REVIEW

**Comment by csrwng** [comment]
@pmorie I see this as the status on the `cluster-federation.v0.0.5` csv: 
```
SingleNamespace InstallModeType not supported, cannot configure to watch one namespace
```

**Comment by pmorie** [comment]
/wip

This is still wip, I'm working to debug it now.

File: manifests/cluster-federation/0.0.5/cluster-federation.v0.0.5.clusterserviceversion.yaml [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+  annotations:
+    capabilities: Basic Install
+    categories: "OpenShift Optional, Integration & Delivery"
+    description: Cluster-scoped Federation
+    certified: "false"
+    containerImage: "quay.io/openshift/origin-federation-controller:v4.0.0"
+    createdAt: "2019-01-01T00:00:00Z"
+spec:
+  displayName: Cluster Federation
+  description: |
```
Comment (pmorie) [inline manifests/cluster-federation/0.0.5/cluster-federation.v0.0.5.clusterserviceversion.yaml]: @robszumski can you help figure out how to meaningfully differentiate this from the other CSV?

**Comment by pmorie** [comment]
/retest

...to pick up changes in #46

**Pull request merged**